src/sage/rings/polynomial/generalized_order.py

```diff
@@ -0,0 +1,428 @@
+r"""
+Generalized monomial orders on `\ZZ^n` are total orders on `\ZZ` introduced in
+[PU1999]_ and defined by a triple:
+
+- Conic decomposition: a finite covering of the additive monoïd `\ZZ^n` by submonoids.
+  In the current implementation, the conic decomposition of Example 2.2 in 
+  [PU1999]_ is imposed.
+
+- Group order: this can be any class with a 'greatest_tuple' method returning
+  the greatest of two tuples based on a group order on `\ZZ^n`. 
+  Available choices include 'lex'.
```

Comment:
It's not really the case.
However, I think that accepting any such class sounds like a good idea.